tremor-script/src/ast/deploy/raw.rs

```diff
@@ -0,0 +1,816 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// We want to keep the names here
+#![allow(clippy::module_name_repetitions)]
+
+use super::BaseExpr;
+use super::ConnectorDecl;
+use super::CreateStmt;
+use super::FlowDecl;
+use super::PipelineDecl;
+use super::Value;
+use crate::ast::error_generic;
+use crate::ast::query::raw::QueryRaw;
+use crate::ast::raw::ExprRaw;
+use crate::ast::raw::IdentRaw;
+use crate::ast::raw::ModuleRaw;
+use crate::ast::raw::StringLitRaw;
+use crate::ast::raw::WithExprsRaw;
+use crate::ast::AggrRegistry;
+use crate::ast::BaseRef;
+use crate::ast::Deploy;
+use crate::ast::DeployStmt;
+use crate::ast::Expr;
+use crate::ast::Helper;
+use crate::ast::ModDoc;
+use crate::ast::NodeMetas;
+use crate::ast::Registry;
+use crate::ast::Script;
+use crate::ast::StringLit;
+use crate::ast::Upable;
+use crate::errors::ErrorKind;
+use crate::errors::Result;
+use crate::impl_expr;
+use crate::interpreter::Cont;
+use crate::pos::Location;
+use crate::AggrType;
+use crate::EventContext;
+use beef::Cow;
+use halfbrown::HashMap;
+use tremor_common::time::nanotime;
+use tremor_common::url::TremorUrl;
+use tremor_value::literal;
+use value_trait::Mutable;
+
+// For compile time interpretation support
+use crate::interpreter::Env;
+use crate::interpreter::ExecOpts;
+use crate::interpreter::LocalStack;
+use crate::Return;
+
+pub type DeployWithExprsRaw<'script> = Vec<(IdentRaw<'script>, ExprRaw<'script>)>;
+
+/// Evaluate a script expression at compile time with an empty state context
+/// for use during compile time reduction
+/// # Errors
+/// If evaluation of the script fails, or a legal value cannot be evaluated by result
+pub fn run_script<'script, 'registry>(
+    helper: &Helper<'script, 'registry>,
+    expr: &Script<'script>,
+) -> Result<Value<'script>> {
+    // We duplicate these here as it simplifies use of the macro externally
+    let ctx = EventContext::new(nanotime(), None);
+    let mut event = literal!({}).into_static();
+    let mut state = literal!({}).into_static();
+    let mut meta = literal!({}).into_static();
+
+    match expr.run(&ctx, AggrType::Emit, &mut event, &mut state, &mut meta) {
+        Ok(Return::Emit { value, .. }) => Ok(value),
+        // FIXME this deserves a proper error
+        _otherwise => error_generic(
+            expr,
+            expr,
+            &"Failed to evaluate script at compile time".to_string(),
+            &helper.meta,
+        ),
+    }
+}
+
+/// Evaluate an expression at compile time with an empty state context
+/// for use during compile time reduction
+/// # Errors
+/// If evaluation of the expression fails, or a legal value cannot be evaluated by result
+pub(crate) fn run_expr<'script, 'registry>(
```

Comment:
Note from the chat with @yatinmaan before I forget it. In the subquery PR we discovered that using `run` for this is problematic as it will allow evaluating non-const expressions leading to unexpected behaviour, the `reduce` functions are a better fit - modular subqueries has a template on how to do that we can copy here.

Reply:
Yes, that's fine. Once subquery is merged in we can normalize and prep Troy for merging into connectors

Reply:
We can look at this now that SQ is merged and rebased into troy

Reply:
This was resolved with the args/with and subquery/pipeline refactoring. Closing.